sdk/core/core-util/src/createAbortablePromise.ts

```diff
@@ -28,7 +28,7 @@ export function createAbortablePromise<T>(
   const { cleanupBeforeAbort, abortSignal, abortErrorMsg } = options ?? {};
   return new Promise((resolve, reject) => {
     function rejectOnAbort(): void {
-      reject(new AbortError(abortErrorMsg ?? "The operation was aborted."));
+      reject(new AbortError(abortSignal?.reason ?? abortErrorMsg ?? "The operation was aborted."));
```

Comment:
This is getting a little complex, I wonder if we can just go ahead and replace our AbortError and all that logic with `throwIfAborted` or other native APIs, now that we are on node 18 and higher https://developer.mozilla.org/en-US/docs/Web/API/AbortSignal#browser_compatibility

Reply:
let's also make sure to not introduce regressions or breaking changes :P